### PR TITLE
Add more details about Firebase hosting

### LINKED ISSUE #6770: Add more details about firebase hosting to 'Build and release a web app' page
### Page URL

https://docs.flutter.dev/deployment/web/

### Page source

https://github.com/flutter/website/tree/main/src/deployment/web.md

### Describe the problem

The end of the page only briefly mentions using GitHub or Firebase for hosting. We should do more!

### Expected fix

Likely need a new page here. I started sketching out some content here – at least useful for inspiration - https://github.com/kevmoo/flutter_web_firebase_hosting/

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by johnpryan**
This also organizes the sections in order of usefulness, first building the app for release, then deploying to the web, then deploying to Firebase Hosting.

closes #6770

### CODE REVIEW

File: src/deployment/web.md
Comment (sfshaza2) [inline src/deployment/web.md:68]:
```suggestion
  The `canvaskit` directory and its contents are only present when the
  CanvasKit renderer is selected—not when the HTML renderer is selected.
```

File: src/deployment/web.md
Comment (sfshaza2) [inline src/deployment/web.md:243]: Case matters for links.

```suggestion
[install-firebase-cli]: https://firebase.google.com/docs/cli#install_the_firebase_cli
```

File: src/deployment/web.md
Comment (sfshaza2) [inline src/deployment/web.md:117]:
```suggestion
3. Answer `yes` when asked if you want to use a web framework.
```

File: src/deployment/web.md
Comment (sfshaza2) [inline src/deployment/web.md:120]:
```suggestion
4. If you're in an empty directory,
    you'll be asked to choose your web framework. Choose `Flutter Web`.
```

File: src/deployment/web.md
Comment (sfshaza2) [inline src/deployment/web.md:122]:
```suggestion
5. Choose your hosting source directory; this could be an existing flutter app.
```

File: src/deployment/web.md
Comment (sfshaza2) [inline src/deployment/web.md:134]:
```suggestion
    Running this command automatically runs `flutter build web --release`,
```

**Review by sfshaza2 — APPROVED** [review]
LGTM, once some minor nits are merged.

File: src/deployment/web.md
Comment (atsansone) [inline src/deployment/web.md:68]: **issue:** Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).  Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
When you select the CanvaKit renderer, the `canvaskit` directory
and its contents appear. When you select the HTML renderer,
this directory doesn't appear.
```

File: src/deployment/web.md
Comment (atsansone) [inline src/deployment/web.md:138]: **nit:** Phrase as a call to action.
```suggestion
To learn more, visit the official [Firebase Hosting][] documentation for
Flutter on the web.
```

File: src/deployment/web.md
Comment (atsansone) [inline src/deployment/web.md:146]: **issues:** Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading). Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
The web supports the standard `Image` widget to display images.
By design, web browsers run untrusted code without harming the host computer.
This limits what you can do with images compared to mobile and desktop platforms.

```

File: src/deployment/web.md
Comment (atsansone) [inline src/deployment/web.md:155]: **nit:** Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
CanvasKit on desktop browsers. We recommend this combination
to optimize for the characteristics of each platform.
```

File: src/deployment/web.md
Comment (atsansone) [inline src/deployment/web.md:171]: **suggestion:** Simplify as a table:
```suggestion
| Type of web app build | Code minified? | Tree shaking performed? |
|-----------------------|----------------|-------------------------|
| debug                 | No             | No                      |
| profile               | No             | Yes                     |
| release               | Yes            | Yes                     |
```

File: src/deployment/web.md
Comment (atsansone) [inline src/deployment/web.md:130]: **chore:** Set this to `terminal`.
```suggestion
    ```terminal
```

File: src/deployment/web.md
Comment (atsansone) [inline src/deployment/web.md:135]: **issue:** 
Write as shorter sentences per [GSG](https://developers.google.com/style/accessibility#ease-of-reading).
```suggestion
    By running this command, Flutter invokes `flutter build web --release`.
    You don't have to build your app in a separate step.
```

File: src/deployment/web.md
Comment (atsansone) [inline src/deployment/web.md:138]: **nit:** Recommend a call to action instead.
```suggestion
To learn more, visit the official [Firebase Hosting][] documentation for
Flutter on the web.
```

**Review by atsansone — COMMENTED** [review]
@johnpryan : Recommend a little tightening of copy.

**Review by sfshaza2 — APPROVED** [review]
lgtm!

**Pull request merged**
